src/lib/models/subscriptions.rs

```diff
@@ -0,0 +1,126 @@
+use serde_json::Value as JsonValue;
+use sqlx::{query_as, Error, PgPool};
+use time::OffsetDateTime;
+use uuid::Uuid;
+
+#[derive(Debug)]
+pub struct Subscription {
+    pub id: Uuid,
+    pub flow_id: String,
+    pub subscription_id: String,
+    pub report_timestamp: OffsetDateTime,
+    pub subscription_created: OffsetDateTime,
+    // Note this is a hash
+    pub fxa_uid: String,
+    pub quantity: i32,
+    pub plan_id: String,
+    pub plan_currency: String,
+    pub plan_amount: i32,
+    pub country: Option<String>,
+    pub aic_id: Option<Uuid>,
+    pub aic_expires: Option<OffsetDateTime>,
+    pub cj_event_value: Option<String>,
+    pub status: Option<String>,
+    pub status_history: Option<JsonValue>,
+}
+impl PartialEq for Subscription {
+    fn eq(&self, other: &Self) -> bool {
+        let simple_match = self.id == other.id &&
+        self.flow_id == other.flow_id &&
+        self.subscription_id == other.subscription_id &&
+        // When timestamps go in and out of database they lose precision to milliseconds
+        self.report_timestamp.millisecond() == other.report_timestamp.millisecond() &&
+        self.subscription_created.millisecond() == other.subscription_created.millisecond() &&
+        self.fxa_uid == other.fxa_uid &&
+        self.quantity == other.quantity &&
+        self.plan_id == other.plan_id &&
+        self.plan_currency == other.plan_currency &&
+        self.plan_amount == other.plan_amount &&
+        self.country == other.country &&
+        self.aic_id == other.aic_id &&
+        self.cj_event_value == other.cj_event_value &&
+        self.status == other.status
+        // Compare manually if needed
```

Comment:
What does this mean?

Reply:
This means that I can say does subA == subB and it won't match on status_history. So if I really need to match on status history (this is all for testing) i can do that manually.